pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/v2/PNConfiguration.kt

```diff
@@ -59,8 +59,11 @@ interface PNConfiguration : com.pubnub.api.v2.PNConfiguration {
         val secretKey: String
 
         /**
-         * If Access Manager is utilized, client will use this authKey in all restricted requests.
+         * If Access Manager (deprecated PAM v2) is utilized, client will use this authKey in all restricted requests.
          */
+        @Deprecated(
+            "This setting is deprecated because it relates to deprecated Access Manager (PAM V2). Use method pubnub.setToken(token) instead.",
```

Comment:
IMHO in all the new documentation strings we're missing the opportunity to educate the user about the difference between v2 and v3. In all your changes in this PR, the call to action for the user is to take the PAM v2 authKey and set it through pubnub.setToken. But in reality, they should migrate from v2 to v3 by going to https://www.pubnub.com/docs/general/resources/migration-guides/pam-v3-migration and making the appropriate changes to their infrastructure. I think we should link to the guide and be more explicit about using v3 token instead of authKey